kong/cmd/utils/nginx_signals.lua

```diff
@@ -101,6 +147,11 @@ function _M.start(kong_conf)
     return nil, "nginx is already running in " .. kong_conf.prefix
   end
 
+  local has_process_secrets, err = set_process_secrets_env(kong_conf)
```

Comment:
So lets say nginx crashes ungracefully, we'd have a encrypted & serialized `KONG_PROCESS_SECRETS` environment variable on the system. Could we think of any way to abuse that?

Reply:
@jschmid1, there is a slight chance that secrets are visible for short amount of time on `ENV` or in `.kong_process_secrets` file. They are in encrypted form, but encryption key is artificial and can be generated. We can think that someone gets access to env var or the file and can then decrypt it by calculating a key. The CLI (OpenResty timer) sets these and Kong server clears/deletes them on `init`. I am not sure how to make this more robust.

Reply:
@jschmid1, all the ideas are welcomed though. The biggest question is: how to pass secrets from process A to process B in secure enough fashion. We need to do it this way as on server we don't have co-sockets, and would then need to revert to possible incompatible LuaSocket based compatibility layer.

Most of the security should come from the fact that users use a properly audited `vault` so those secrets are not spread from there (e.g. in git repo) and only resolved when the operators start kong. There are bigger issues to solve, I think. Like should `kong.configuration.XXX` show secrets in `plain`? We need that for now, but perhaps as we add rotation, we can remove the `dereferencing` from some fields.

Reply:
> how to pass secrets from process A to process B in secure enough fashion.

I can't think of another way either. Ideally we do rolling-restarts in a controlled environment (which we do not have) where we can pass context between the old and new process. Not an option here unfortunately.

> there is a slight chance that secrets are visible for short amount of time on ENV or in .kong_process_secrets file. They are in encrypted form, but encryption key is artificial and can be generated. 

There should be enough time for the user to initiate key-rotation.